core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -94,7 +95,9 @@ private[spark] class DiskBlockManager(
       } else {
         val newDir = new File(localDirs(dirId), "%02x".format(subDirId))
         if (!newDir.exists()) {
-          Files.createDirectory(newDir.toPath)
+          // SPARK-37618: Create dir as group writable so files within can be deleted by the
+          // shuffle service
+          createDirWithPermission770(newDir)
```

Comment:
Should I add a fallback here in case the ProcessBuilder based approach doesn't work, since this is a more widely used codepath than the merged based shuffle? Not sure if there's specific operating systems or environments that it always won't work in